### PR TITLE
Adds util to compare Metadata for remote state

### PR DESCRIPTION
**Pull request by linuxpi**
### Description
- Adds a new util to compare Metadata content(excluding IndexMetadata) for entities uploaded to remote store as part of remote state.
- When cluster uuid trimming logic runs, it would use this util to compare metadata b/w two different UUIDs

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/10662

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/cluster/metadata/Metadata.java [hunk 1]
```diff
@@ -922,19 +922,26 @@ public static boolean isGlobalStateEquals(Metadata metadata1, Metadata metadata2
...
             return false;
         }
-        if (!metadata1.clusterUUID.equals(metadata2.clusterUUID)) {
+        return isRemoteGlobalStateEquals(metadata1, metadata2);
+    }
+
+    /**
+     * Compares Metadata entities persisted in Remote Store.
+     */
+    public static boolean isRemoteGlobalStateEquals(Metadata metadata1, Metadata metadata2) {
```
Comment (shwetathareja) [inline server/src/main/java/org/opensearch/cluster/metadata/Metadata.java:940]: isGlobalResourcesMetadataEquals lets not tie it with remote

**Comment by shwetathareja** [comment]
Add UTs

File: server/src/main/java/org/opensearch/cluster/metadata/Metadata.java (hunk 1 above)
Comment (linuxpi) [reply server/src/main/java/org/opensearch/cluster/metadata/Metadata.java]: updated

**Comment by linuxpi** [comment]
Flaky test 

[org.opensearch.repositories.azure.AzureBlobContainerRetriesTests.testWriteLargeBlob](https://build.ci.opensearch.org/job/gradle-check/28549/testReport/junit/org.opensearch.repositories.azure/AzureBlobContainerRetriesTests/testWriteLargeBlob/)

File: server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java [hunk 2]
```diff
@@ -882,7 +896,7 @@ public void testGetValidPreviousClusterUUIDWithMultipleChains() throws IOExcepti
             "cluster-uuid3",
             "cluster-uuid1"
         );
-        mockObjectsForGettingPreviousClusterUUID(clusterUUIDsPointers);
+        mockObjectsForGettingPreviousClusterUUID(clusterUUIDsPointers, randomBoolean());
```
Comment (shwetathareja) [inline server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java:899]: if this is random, how would you ensure previous matches uuid-3 for the assertion below.

**Review by shwetathareja — APPROVED** [review]

**Pull request merged**